include/swift/Remote/CMemoryReader.h

```diff
@@ -62,17 +62,31 @@ class CMemoryReader final : public MemoryReader {
     if (!length)
       return false;
 
-    auto buffer = std::unique_ptr<uint8_t>(new uint8_t[length + 1]);
-    if (!readBytes(address, buffer.get(), length + 1))
+    const void *ptr;
+    std::function<void()> freeFunc;
+    std::tie(ptr, freeFunc) = readBytes(address, length);
+    if (ptr != nullptr) {
+      dest = std::string(reinterpret_cast<const char *>(ptr), length);
+      freeFunc();
+      return true;
+    } else {
       return false;
-
-    dest = std::string(reinterpret_cast<const char *>(buffer.get()));
-    return true;
+    }
   }
 
-  bool readBytes(RemoteAddress address, uint8_t *dest, uint64_t size) override {
-    return Impl.readBytes(Impl.reader_context,
-                          address.getAddressData(), dest, size) != 0;
+  std::tuple<const void *, std::function<void()>>
```

Comment:
Why not just pass back a `std::unique_ptr<uint8_t *>`?

Reply:
Just ignorance/forgetfulness on my part. Good suggestion, I'll change it.